lib/actuator/include/actuator/MotorParameters.hpp

```diff
@@ -0,0 +1,75 @@
+/// @ingroup     actuator
+/// @{
+/// @file
+/// @brief       Configuration structure for initializing a Motor object.
+/// @author      Gilles DOFFE <g.doffe@gmail.com>
+
+#pragma once
+
+// System includes
+#include <cstdint>
+
+// RIOT includes
+#include <motor_driver.h>
+#include <periph/gpio.h>
+
+// Project includes
+#include "Actuator.hpp"
+#include "odometer/OdometerInterface.hpp"
+#include "motor/MotorInterface.hpp"
+// Motion control
+#include "motor_pose_filter/MotorPoseFilterParameters.hpp"
+#include "pose_pid_controller/PosePIDControllerParameters.hpp"
+#include "speed_pid_controller/SpeedPIDControllerParameters.hpp"
+#include "speed_filter/SpeedFilterParameters.hpp"
+
+namespace cogip {
+namespace actuators {
+namespace positional_actuators {
+
+/// @struct MotorParameters
+/// @brief Configuration structure for initializing a Motor object.
+/// @details
+///   Gathers all necessary parameters to configure and instantiate a Motor,
+///   including control, filtering, timeout, and hardware interface options.
+struct MotorParameters {
+    /// @brief Motor identifier (enum used internally to differentiate motors).
+    cogip::actuators::Enum id;
+
+    /// @brief Default timeout period, in milliseconds.
+    /// @details Used to determine if the motor should be disabled due to inactivity.
+    uint32_t default_timeout_ms = 0;
+
+    /// @brief Callback used to send the actuator state.
+    send_state_cb_t send_state_cb = nullptr;
+
+    /// @brief GPIO pin used to clear the motor's overload state.
+    gpio_t clear_overload_pin = GPIO_UNDEF;
+
+    /// @brief Parameters for the PosePIDController (position control).
+    cogip::motion_control::PosePIDControllerParameters* pose_controller_parameters = nullptr;
+
+    /// @brief Parameters for the SpeedPIDController (speed control).
+    cogip::motion_control::SpeedPIDControllerParameters* speed_controller_parameters = nullptr;
+
+    /// @brief Parameters for the MotorPoseFilter (used to smooth and validate position).
+    cogip::motion_control::MotorPoseFilterParameters* motor_pose_filter_parameters = nullptr;
+
+    /// @brief Parameters for the SpeedFilter (used to smooth and limit speed).
+    cogip::motion_control::SpeedFilterParameters* speed_filter_parameters = nullptr;
```

Comment:
I think the whole controllers parameters pointers can be replaced by references ? As we always need to define it.

Reply:
Done. I think it could also be replaced by references inside the motion_control library but I prefer to keep that modification for motion control incoming rework.